proto/connector/v1/connector.proto

```diff
@@ -54,6 +54,27 @@ service SourcePlugin {
   // other function. After Teardown returns, the plugin should be ready for a
   // graceful shutdown.
   rpc Teardown(Source.Teardown.Request) returns (Source.Teardown.Response);
+
+  // -- Lifecycle events -------------------------------------------------------
+
+  // LifecycleOnCreated is called after Configure and before Start when the
+  // connector is run for the first time. This call will be skipped if a
+  // connector was already started before. This method can be used to do some
+  // initialization that needs to happen only once in the lifetime of a
+  // connector (e.g. create a replication slot). Anything that the connector
+  // creates in this method is considered to be owned by this connector and
+  // should be cleaned up in LifecycleOnDeleted.
+  rpc LifecycleOnCreated(Source.Lifecycle.OnCreated.Request) returns (Source.Lifecycle.OnCreated.Response);
```

Comment:
I added the prefix `Lifecycle` to the methods to make them stand out a bit from the other methods. It's a bit verbose, but hopefully makes the code clearer. Let me know what you think.

Reply:
wdyt about `LifecycleCreated`? does it take away from the intended meaning?

Reply:
That sounds to me like a lifecycle was created (or updated or deleted) 😅

Reply:
ah makes sense!

Reply:
At least to me, `OnCreate` is clear enough, and most frameworks I've at least used use such naming. However, I don't have anything against adding the Lifecycle prefix either.